kernel/arch/riscv/paging.c

```diff
@@ -126,21 +482,128 @@ map_pages_int(pdir_t *pdir,
               bool big_pages_allowed,
               u32 hw_flags)
 {
-   NOT_IMPLEMENTED();
+   int rc;
+   size_t pages = 0;
+   size_t big_pages = 0;
+   size_t rem_pages = page_count;
+   u32 big_page_flags;
+
+   ASSERT(!((ulong)vaddr & OFFSET_IN_PAGE_MASK));
+   ASSERT(!(paddr & OFFSET_IN_PAGE_MASK));
+
+   if (big_pages_allowed && rem_pages >= PTRS_PER_PT) {
+
+      for (; pages < rem_pages; pages++) {
+
+         if (!((ulong)vaddr & (L1_PAGE_SIZE - 1)) &&
```

Comment:
This is hard to read and we'll need in other places to check for the alignment against L1 pages. Can you introduce a nice macro such `PAGING_IS_L1_ALIGNED()` or something like that? You can totally pick a better name, of course.

Reply:
fixed!